scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -198,17 +198,20 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         i
       }
       val next = if (nextPos != -1) scannerTokens(nextPos) else null
-      def nextScannerToken: Option[Token] =
-        if (currPos + 1 < scannerTokens.length) Some(scannerTokens(currPos + 1))
-        else None
+      def lineEndsAfterComma: Boolean = {
+        var i = currPos + 1
+        val end = Math.min(scannerTokens.length - 1, nextPos - 1)
+        while (i < end && !scannerTokens(i).is[LineEnd]) i += 1
+        scannerTokens(i).is[LineEnd]
+      }
```

Comment:
PS I've been programming in Scala for 5 years, and this is the first while loop I've written 😅

Reply:
> PS I've been programming in Scala for 5 years, and this is the first while loop I've written 😅

hehe, while loops are sometimes the clearest way to write an algorithm IMO, in particular graph traversals.

I am a bit confused by the logic in this method since it relies on knowledge outside of the method. The implementation doesn't seem related to commas. I was thinking something more like this
```scala
@tailrec
def nextScannerToken(i: Int): Option[Token] =
  if (i >= scannerTokens.length) None
  else {
    val next = scannerTokens(i)
    if (next.is[Comment]) nextScannerToken(i + 1)
    else Some(next)
  }
...
nextScannerToken(currPos + 1)
```

Reply:
Yes, my implementation relies on previous computations. It could actually be a `val`, ora a `lazy val`. The only reason it's a `def` is that it doesn't make sense to compute it for another case other than when `curr.is[Comma] && next.is[CloseDelim]` is `true`.

It's not meant as a generic utility method. I could refactor it like this to make it clearer:

```scala
def isTrailingComma: Boolean =
  if (dialect.allowTrailingCommas && curr.is[Comma] && next.is[CloseDelim]) {
    var i = currPos + 1
    val end = Math.min(scannerTokens.length - 1, nextPos - 1)
    while (i < end && !scannerTokens(i).is[LineEnd]) i += 1
    scannerTokens(i).is[LineEnd]
  } else false
```

which has the advantage of one less method call and not using extra boxing due to `Option`.

Does it seem reasonable?